Hosting Overview: Adjust price style

xavier-lc opened the pull request:
Related to https://github.com/Automattic/dotcom-forge/issues/6384

## Proposed Changes

* This PR adjusts the price styles to make sure it doesn't look misaligned, which happened erratically, as can be seen on this comment: https://github.com/Automattic/wp-calypso/pull/89476#issuecomment-2059177300

## Testing Instructions

* Open the live preview
* Go to `/hosting-overview`
* Check that the plan price card matches the desktop (PZtnZ1nHhnBrsuXDoMJHPD-fi-446%3A153354) and mobile (IQgFHqGiN0eXkPn7Qkth4s-fi-6%3A8860) designs

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Commit:
Adjust price style

Commit:
Remove unneeded `font-style: normal;`

Commit:
Adjust margin

Commit:
Adjust price style on mobile

Commit:
Adjust header style on mobile

Commit:
Adjust price style

rcrdortiz commented:
The design does not match the Figma file, is the design outdated?
I've noticed that the price and `per month` text aren't aligned at a certain width, you can see that it "drops" lower than the number in this case.

![Screenshot 2024-04-25 at 16 02 27](https://github.com/Automattic/wp-calypso/assets/1989914/d455f79a-0439-4f02-86c5-ee162cbd0864)

rcrdortiz commented:
On mobile, there is no left/right space as shown in the Figma file.

Figma:
![Screenshot 2024-04-25 at 16 06 47](https://github.com/Automattic/wp-calypso/assets/1989914/8632cb63-3529-4ca3-889b-f7db73c29640)

Implemented:
![Screenshot 2024-04-25 at 16 06 05](https://github.com/Automattic/wp-calypso/assets/1989914/c01654a7-ac04-41d0-9fc8-ecbad0801213)

Commit:
Use custom price term

The two lines used on `per month` make it difficult to adjust to the mobile view

xavier-lc commented:
> The design does not match the Figma file, is the design outdated?
> I've noticed that the price and `per month` text aren't aligned at a certain width, you can see that it "drops" lower than the number in this case.

I wanted to reuse the `PlanPrice` component, but the two lines used on `per month` make it difficult to adjust to the mobile view, so I dropped it on https://github.com/Automattic/wp-calypso/pull/89895/commits/ed8ed365376cf48c95caff0e339650f102837e89 to follow the designs.

xavier-lc commented:
> On mobile, there is no left/right space as shown in the Figma file.

Yeah, I didn't add it because I thought it would be a padding once the global view was in place, and it's been implemented this way. You can check it adding `/sites?flags=layout%2Fdotcom-nav-redesign-v2` to the live preview URL:

![image](https://github.com/Automattic/wp-calypso/assets/8511199/85070487-b534-4606-8ebb-80877366f5e9)

davemart-in commented:
We can probably pull out this second "per month":

![CleanShot 2024-04-25 at 15 16 03@2x](https://github.com/Automattic/wp-calypso/assets/5634774/ad2e475e-8e2e-4396-9332-abffafd23ef5)

Right now, to me it reads "$8/mo per month"

Commit:
Fix copy

xavier-lc commented:
> Right now, to me it reads "$8/mo per month"

Fixed on https://github.com/Automattic/wp-calypso/pull/89895/commits/b2acde1a262349ce3cac51027edc82be5d7f02c1
